erlang/pop_protocol/pop_protocol.erl

```diff
@@ -30,6 +25,28 @@ compute_transaction_hash(Transaction) when is_map(Transaction) ->
 	 },
     my_crypto:hash( my_serializer:serialize_object(CleanTransaction) ).
 
+check_transaction_correctness(Transaction, ChainId) when is_map(Transaction) ->
+    #{
+      consensus_data := CD,
+      player_id := PlrKey
+     } = Transaction,
```

Comment:
Does this crash if keys don't exist or if Transaction isn't a map (ignoring the fact there is is_map(...))

Reply:
YA :O 
(yes for both)

Reply:
D: